nomad/search_endpoint.go

```diff
@@ -115,10 +115,14 @@ func (s *Search) PrefixSearch(args *structs.SearchRequest,
 
 			for _, ctx := range contexts {
 				iter, err := getResourceIter(ctx, roundUUIDDownIfOdd(args.Prefix, args.Context), ws, state)
+
+				// When searching all Contexts, Job ids will cause errors when searched
+				// in the context of allocs, nodes, and/or evals.
 				if err != nil {
-					return err
+					s.srv.logger.Printf("[WARN] nomad.resources: error when searching context %s for id %s", ctx, args.Prefix)
```

Comment:
This should probably be handled. We shouldn't log an error when it is happy path.